data-prepper-plugins/log-generator-source/src/main/java/com/amazon/dataprepper/plugins/source/loggenerator/logtypes/CommonApacheLogTypeGenerator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.loggenerator.logtypes;
+
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.plugins.source.loggenerator.LogTypeGenerator;
+import com.github.javafaker.Faker;
+import com.github.javafaker.Internet;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@DataPrepperPlugin(name = "apache_clf", pluginType = LogTypeGenerator.class)
+public class CommonApacheLogTypeGenerator implements LogTypeGenerator {
+
+    private final String[] HTTP_METHODS = new String[] {"GET", "POST", "DELETE", "PUT"};
+    private final String[] HTTP_VERSIONS = new String[] {"0.9", "1.0", "1.1", "2.0"};
+    private final String[] FAKE_URIS = new String[] {"/list", "/explore", "/search/tag/list", "/apps/cart.jsp?appID="};
+    private final String[] FAKE_STATUS = new String[] {"200", "404", "500", "301"};
+    private static final String APACHE_COMMON_LOG_FORMAT = "%s %s %s [%s] \"%s %s HTTP/%s\" %s %s";
+    private final Faker faker = new Faker();
+    private final Random random = new Random();
+    private final DateFormat dateFormat = new SimpleDateFormat("dd/MMM/y:HH:mm:ss Z");
+
+    @Override
+    public Event generateEvent() {
+        final Internet internet = faker.internet();
+        final String username = faker.name().username();
+        final String password = internet.password();
+        final String httpMethod = HTTP_METHODS[random.nextInt(4)];
```

Comment:
Please `HTTP_METHODS.size()` instead of `4`. These could easily change and the constant might be missed.